File: ManleyProject3/app/src/main/java/com/example/manleyproject3/MainActivity.java
package com.example.manleyproject3;
import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.TextView;
import android.widget.Button;
import android.widget.EditText;
import android.database.sqlite.SQLiteDatabase;
import android.database.Cursor;
import android.widget.Toast;
import android.content.Intent;
import android.Manifest;
import androidx.core.app.ActivityCompat;

public class MainActivity extends AppCompatActivity{
    private UserDatabaseHelper userDbHelper;
    private EditText usernameEdit;
    private EditText passwordEdit;
    private Button loginButton;
    private Button registerButton;
    private static final int SMS_PERMISSION_CODE = 101;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.login_layout);

        // create user database helper
        userDbHelper = new UserDatabaseHelper(this);
        // assign editTexts and buttons to variables
        usernameEdit = findViewById(R.id.username);
        passwordEdit = findViewById(R.id.password);
        loginButton = findViewById(R.id.loginButton);
        registerButton = findViewById(R.id.registerButton);

        // listener for register button
        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                registerUser();
            }
        });

        // listener for login button
        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                loginUser();
            }
        });
    }

    private void registerUser() {
        // get text from fields
        String username = usernameEdit.getText().toString().trim();
        String password = passwordEdit.getText().toString().trim();

        // handle empty fields
        if (username.isEmpty() || password.isEmpty()) {
            Toast.makeText(this, "Username and password cannot be empty", Toast.LENGTH_SHORT).show();
            return;
        }
        try {
            // add username/password combo to user database
            SQLiteDatabase db = userDbHelper.getWritableDatabase();
            db.execSQL("INSERT INTO users (username, password) VALUES (?, ?)", new Object[]{username, password});
            db.close();
            Toast.makeText(this, "Registration successful!", Toast.LENGTH_SHORT).show();
            // error message
        } catch (Exception e) {
            Toast.makeText(this, "Registration failed: " + e.getMessage(), Toast.LENGTH_SHORT).show();
        }
    }

    private void loginUser() {
        // get text from fields
        String username = usernameEdit.getText().toString().trim();
        String password = passwordEdit.getText().toString().trim();

        // handle empty fields
        if (username.isEmpty() || password.isEmpty()) {
            Toast.makeText(this, "Username and password cannot be empty", Toast.LENGTH_SHORT).show();
            return;
        }

        // search user database for provided username/password combo
        SQLiteDatabase db = userDbHelper.getReadableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM users WHERE username = ? AND PASSWORD = ?", new String[]{username, password});

        // if valid combo move to inventory screen
        if (cursor.moveToFirst()) {
            Toast.makeText(this, "Login successful!", Toast.LENGTH_SHORT).show();
            Intent intent = new Intent(MainActivity.this, Inventory.class);
            startActivity(intent);
        }
        // failed login message
        else {
            Toast.makeText(this, "Invalid username or password", Toast.LENGTH_SHORT).show();
        }
        cursor.close();
        db.close();
    }
}

File: EnhancedArtifact/app/src/main/java/com/example/enhancedartifact/MainActivity.kt
package com.example.enhancedartifact

import android.Manifest
import android.content.Intent
import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.runtime.Composable
import androidx.core.app.ActivityCompat
import com.example.enhancedartifact.ui.theme.LoginScreen

// Login Screen activity
class MainActivity : ComponentActivity() {

    // declare database helper & sms permission code
    private lateinit var userDbHelper: UserDatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        // loads previous instance of the activity
        super.onCreate(savedInstanceState)

        // initializes database helper
        userDbHelper = UserDatabaseHelper(this)

        // loads Compose UI for Login Screen
        setContent {
            LoginScreen(
                onLoginClick = { username, password -> loginUser(username, password) },
                onRegisterClick = { username, password -> registerUser(username, password) }
            )
        }
    }

    // creates a new user on register button press
    private fun registerUser(username: String, password: String) {
        // if username/password is empty
        if (username.isEmpty() || password.isEmpty()) {
            Toast.makeText(this, "Username and password cannot be empty", Toast.LENGTH_SHORT).show()
            return
        }
        // attempt to add username/password to database
        try {
            val db: SQLiteDatabase = userDbHelper.writableDatabase
            db.execSQL("INSERT INTO users (username, password) VALUES (?, ?)", arrayOf(username, password))
            db.close()
            Toast.makeText(this, "registration successful!", Toast.LENGTH_SHORT).show()
        }
        // catch a failure to add account
        catch (e: Exception) {
            Toast.makeText(this, "Registration failed: ${e.message}", Toast.LENGTH_SHORT).show()
        }
    }

    // logs in with entered credentials
    private fun loginUser(username: String, password: String) {
        // if username/password is empty
        if (username.isEmpty() || password.isEmpty()) {
            Toast.makeText(this, "Username and password cannot be empty", Toast.LENGTH_SHORT).show()
            return
        }
        // search entered username/password
        val db: SQLiteDatabase = userDbHelper.readableDatabase
        val cursor = db.rawQuery("SELECT * FROM users WHERE username = ? AND password = ?", arrayOf(username, password))

        // if username/password are in database, log in
        if (cursor.moveToFirst()) {
            val userId = cursor.getInt(cursor.getColumnIndexOrThrow("id"))
            Toast.makeText(this, "Login successful!", Toast.LENGTH_SHORT).show()
            val intent = Intent(this, Inventory::class.java)
            intent.putExtra("USER_ID", userId)
            startActivity(intent)
        }
        // else send failure message
        else {
            Toast.makeText(this, "Invalid username or password", Toast.LENGTH_SHORT).show()
        }
        cursor.close()
        db.close()
    }
}